pkg/types/kustomization.go

```diff
@@ -56,8 +56,15 @@ type Kustomization struct {
 
 	// An Patch entry is very similar to an Resource entry.
 	// It specifies the relative paths for patch files within the package.
-	// URLs and globs are not supported
-	Patches []string `json:"patches,omitempty" yaml:"patches,omitempty"`
+	// URLs and globs are not supported.
+	// The patch files should be Stategic Merge Patch, the default patching behavior for kubectl.
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/strategic-merge-patch.md
+	Patches               []string `json:"patches,omitempty" yaml:"patches,omitempty"`
+	PatchesStrategicMerge []string `json:"patchesSrategicMerge,omitempty" yaml:"patchesStategicMerge,omitempty"`
```

Comment:
Can we define a type
```
type PatchStrategicMerge string
```
and use it here?

It's the same as `string`, but just a different name for better type safety and consistency with other patch types.